presentation/src/main/java/me/androidbox/presentation/event/viewmodel/EventViewModel.kt

```diff
@@ -235,6 +293,7 @@ class EventViewModel @Inject constructor(
 
     private fun verifyVisitorEmail(visitorEmail: String) {
         viewModelScope.launch {
+            onEventScreenEvent(EventScreenEvent.OnVerifyingVisitorEmail(true))
```

Comment:
I noticed you create a ton of events that aren't really necessary

This "OnVerifyingVisitorEmail" is a method that's only being used on this verifyVisitorEmail method

So you can remove that event, and replace it with the state updates 
```
_eventScreenState.update { eventScreenState ->
                    eventScreenState.copy(isVerifyingVisitorEmail = true)
                }
```
and
```
_eventScreenState.update { eventScreenState ->
                    eventScreenState.copy(isVerifyingVisitorEmail = false)
                }
```